wpilibj/src/main/java/edu/wpi/first/wpilibj/LEDPattern.java

```diff
@@ -93,6 +93,18 @@
  */
 @FunctionalInterface
 public interface LEDPattern {
+  /** A functional interface for index mapping functions. */
+  @FunctionalInterface
+  interface IndexMapper {
+    /**
+     * Maps the index.
+     *
+     * @param bufLen Length of the buffer
+     * @param index The index to map
+     */
+    int apply(int bufLen, int index);
+  }
```

Comment:
Why not use `java.util.function.IntBinaryOperator`?

Reply:
I thought that using `IntBinaryOperator` might confuse things, since the name of `BinaryOperator` usually implies that the two parameters are more or less the same. I think autocomplete on IDEs would also tend to fill in `(left, right) -> {}` for an `IntBinaryOperator` as the parameter names, which also isn't helpful. Doesn't matter too much, but I figured might as well.